[bryanjhc] iP

bryanjhc opened the pull request:
## JackBean
> "The homie you can always count on to keep track of your tasks!" ~ Bryan Chew

JackBean is a task tracker that helps you keep track of *3 different types* of tasks:
- Todos
- Deadlines:
- Events

To run it, all you need to do are the following:
1. Download the code from [my repo](github.com/bryanjhc/ip).
2. Compile the Codes
3. Run `jackbean.gui.Launcher.main()` or use `gradlew run`
4. Type `help` in the input
5. Use any command from the list of commands given!

IT'S ~~P2W~~ **F2P** 😄

Features implemented so far:
- [x] Task tracking
- [x] Search using keyword
- [x] GUI

Here's a snippet of the code for `JackBean.main()`:
```Java
public static void main(String[] args) {
    Ui.greet();
    TaskList taskList = new TaskList();
    Storage.fetchStorage(taskList);


    Scanner userInput = new Scanner(System.in);
    // GitHub CoPilot helped a lot with auto-complete (except personalisation)
    // it is actually really smart haha
    while (true) {
        String input = userInput.nextLine();

        // handle bye
        if (input.equalsIgnoreCase("bye")) { // I added equalsIgnoreCase() by myself
            Ui.sayGoodBye();
            userInput.close();
            break;
        }

        // handle other input
        try {
            String reply = Parser.parseUserInput(input, taskList);
            Ui.showMessage(reply);
        } catch (Exception e) {
            Ui.showMessage("Homie! There was an error:\n" + e.toString());
        }
    }

}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

bryanjhc committed:
renamed Duke to JackBean, did greetings and exit, added AI.md file

bryanjhc committed:
declared and defined greeting and exitMessage strings for use in main()

bryanjhc committed:
added echo

bryanjhc committed:
added list functionality

bryanjhc committed:
added marking as done and task class

bryanjhc committed:
added subclasses of tasks

bryanjhc committed:
extra commit to tag A-Classes which wasnt tagged in previous commits

bryanjhc committed:
added logic from adding the different subclasses of tasks

bryanjhc committed:
added auto ui testing

bryanjhc committed:
added two checked exceptions and an enum class for user input validation

bryanjhc committed:
added invalid task type exception

bryanjhc committed:
finished user input validation basic

bryanjhc committed:
changed EXPECTED.TXT

bryanjhc committed:
added the delete logic and edit test cases

bryanjhc committed:
added an update on AI.md

bryanjhc committed:
added help instructions for delete

bryanjhc committed:
Add save functionality via ./data/JackBeanStorage.txt

bryanjhc committed:
Merge branch 'branch-Level-7'

bryanjhc committed:
Allow date recognition by JackBean ChatBot

bryanjhc committed:
Merge branch 'branch-Level-8'

bryanjhc committed:
Add more OOP classes according to A-MoreOOP

bryanjhc committed:
Add Ui Class to handle User Interface

bryanjhc committed:
Merge branch 'branch-A-MoreOOP'

bryanjhc committed:
Improve OOP usage of Ui Class

bryanjhc committed:
Add packaging for JackBean ChatBot

bryanjhc committed:
Merge branch 'branch-A-Packages'

bryanjhc committed:
Merge branch 'add-gradle-support'

bryanjhc committed:
Add gradle support

bryanjhc committed:
Add JUnit Testing

bryanjhc committed:
Merge branch 'branch-A-JUnit'

bryanjhc committed:
Add JavaDoc comments for all classes and non-trivial methods

bryanjhc committed:
Merge branch 'branch-A-JavaDoc'

bryanjhc committed:
Comply with the CS2103T Coding Standard

bryanjhc committed:
Merge branch 'branch-A-CodingStandard'

bryanjhc committed:
Add find functionality

bryanjhc committed:
Merge branch 'branch-Level-9'

CJianzhi commented on src/main/java/jackbean/command/Ui.java:
```diff
@@ -0,0 +1,44 @@
+ * Represents the user interface of the JackBean chatbot.
+ */
+public class Ui {
+    private static final String horizontalLine = "____________________________________________________________";
```
Should the name horizontalLine be in capital letters since it is a constant?

bryanjhc replied:
Ooh, thanks for catching that JianZhi!

CJianzhi commented on src/main/java/jackbean/task/Deadline.java:
```diff
@@ -0,0 +1,50 @@
+ * This JavaDoc was written by GitHub Copilot.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
Should the name of this variable be changed to something more descriptive?

bryanjhc replied:
Sure! could deadlineDate work?

CJianzhi commented on src/main/java/jackbean/task/Event.java:
```diff
@@ -0,0 +1,64 @@
+ * This JavaDoc was written by GitHub Copilot.
+ */
+public class Event extends Task {
+    protected LocalDate from;
```
Should the names of these fields be changed to nouns?

bryanjhc replied:
Sure!

CJianzhi commented on src/main/java/jackbean/task/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+     *
+     * @return The number of tasks in the task list.
+     */
+    public String howManyTasks() {
```
Should the name of this method be changed to include an action/verb such as getNumberOfTasks?

CJianzhi commented on src/test/java/jackbean/command/ParserTest.java:
```diff
@@ -0,0 +1,60 @@
+public class ParserTest {
+    TaskList dummyTaskList = new TaskList();
+    @Test
+    public void parseUserInput_invalidCommand_exceptionThrown() {
```
I like how the test method names are easy to understand!

bryanjhc replied:
Thanks bro!

CJianzhi reviewed (commented):
LGTM :thumbsup:. Overall, code is written well with good naming conventions used that allow readers to understand logic of code written. There are only very little nits to fix.

zzibo commented on src/main/java/jackbean/command/Parser.java:
```diff
@@ -9,112 +9,113 @@
+        if (input.equalsIgnoreCase("help")) {
+            // GitHub copilot helped mostly with the response to help message
+            StringBuilder reply = new StringBuilder("Yo homie! Here are the commands you can use:\n");
+            reply.append("1. list - to list all tasks\n");
```
Will it be better to include all these messages in a Constant.java class?

bryanjhc replied:
thats a good suggestion!

zzibo commented on src/main/java/jackbean/command/Parser.java:
```diff
@@ -9,112 +9,113 @@
+            Storage.updateStorage(taskList);
+            return "LESGOOO homie! Good job on finishing this task:\n"
+                    + taskList.getTask(taskNumber);
+        } else if (input.toLowerCase().startsWith("unmark")) {
```
Maybe you can consider using a switch case and make it look cleaner :}

bryanjhc replied:
That makes sense!

zzibo commented on src/main/java/jackbean/command/Parser.java:
```diff
@@ -9,112 +9,113 @@
-                // first use split and then parse the integer
-                String[] splitInput = input.split(" ");
-                int taskNumber = Integer.parseInt(splitInput[1]);
+    public static String parseUserInput(String input, TaskList taskList)
```
Maybe can consider adding java docs for your public methods!

bryanjhc replied:
Sure thing!

zzibo reviewed (commented):
LGTM! Overall the code looks good and the exception handling was well done. A few nit to pick and more abstraction can help make your code look cleaner!

bryanjhc committed:
Make some changes based on the pull request peer reviews

bryanjhc committed:
Integrate basic JavaFX

bryanjhc committed:
Integrate a working GUI

bryanjhc committed:
Integrate FXML

bryanjhc committed:
Add profile images for GUI

bryanjhc committed:
Merge branch 'branch-Level-10'

bryanjhc committed:
Add easter egg commands

bryanjhc committed:
Follow checkstyle requirements

bryanjhc committed:
Add assertions to catch programmer related bugs

Introduced assertions to multiple files of code. This change allows
the program to catch bugs that may occur due to the fault of the
programmer.

bryanjhc committed:
Adhere to Code Quality standards

This change refactors multiple parts of code, with the most common
refactoring being to extract method. This significantly shortens the
parseUserInput method in Parser.java.

bryanjhc committed:
Merge pull request #1 from bryanjhc/branch-A-Assertions

Add assertions to catch programmer related bugs

bryanjhc committed:
Merge branch 'master' into branch-A-CodeQuality

bryanjhc committed:
Merge pull request #2 from bryanjhc/branch-A-CodeQuality

Adhere to Code Quality standards

bryanjhc committed:
Improve search functionality

This change allows the user to search for tasks even if the keyword
matches the item only partially. It also allows the user to search
by task type and date.

bryanjhc committed:
Add a small change to adhere to Coding Style

bryanjhc committed:
Add personality to the JackBean Chatbot

Personality is given via a custom name, customised reply phrases,
customised profile photos as well as easter egg commands.

bryanjhc committed:
Edit AI.md and README.md

bryanjhc committed:
Add GUI title and Ui.png

bryanjhc committed:
Add documentation for JackBean

bryanjhc committed:
Add javadocs for some methods
